google/cloud/security/scanner/scanner.py

```diff
@@ -69,13 +69,50 @@
                      'the format of the path should be '
                      '"gs://bucket-name/path/for/output".'))
 
+flags.DEFINE_bool('list_engines', False, 'List all rule engines')
+
+flags.DEFINE_string('use_engine', None, 'Which engine to use')
+
+flags.DEFINE_string('use_scanner_basedir', None, 'Which rule basedir to use')
+
 LOGGER = log_util.get_logger(__name__)
 SCANNER_OUTPUT_CSV_FMT = 'scanner_output.{}.csv'
 OUTPUT_TIMESTAMP_FMT = '%Y%m%dT%H%M%SZ'
 
 
 def main(_):
     """Run the scanner."""
+    scanner_base_dir = os.path.dirname(__file__)
+
+    # Allow loading of rules engines and pipelines from different location
+    if not FLAGS.use_scanner_basedir:
+        LOGGER.info("Using %s as scanner base direcotry",
+                    scanner_base_dir)
+    else:
+        scanner_base_dir = FLAGS.use_scanner_basedir
+        LOGGER.info("Using %s as scanner base directory",
+                    scanner_base_dir)
+
+    # Setup base directories
+    audit_base_dir = scanner_base_dir + '/audit/'
+    data_pipeline_base = scanner_base_dir + '/pipelines/'
+
+    if FLAGS.list_engines:
+        _rules_engine_list(audit_base_dir)
+
+    if not FLAGS.use_engine:
+        LOGGER.warn('Provide an engine file')
+        sys.exit(1)
+    else:
+        rules_engine_filename = FLAGS.use_engine
+
+    # Load rules engine according to command line flag
+    rules_engine_class, rules_engine_name = _create_rules_engine(
```

Comment:
Hmmm, perhaps we can create a map and not need the _create_rules_engine()?

For example:
```python
ENGINE_TO_DATA_MAP =  {
    'IamRulesEngine': audit.IamRulesEngine,
    'FooRulesEngine': audit.FooRulesEngine,
    'BarRulesEngine': audit.BarRulesEngine,
}

# So, that we can do something like this:
rules_engine = ENGINE_TO_DATA_MAP['IamRulesEngine'](rules_file_path=FLAGS.rules)
```